### PR TITLE
[v8.4.x] Security: Sync security changes on main (#45083)

### PR DESCRIPTION
**Pull request by vtorosyan**
**What this PR does / why we need it**:

Manual backport of https://github.com/grafana/grafana/pull/45083

### CODE REVIEW

File: pkg/api/pluginproxy/pluginproxy_test.go [hunk 1]
```diff
@@ -245,6 +246,46 @@ func TestPluginProxy(t *testing.T) {
...
+			Context: &web.Context{
+				Req:  httptest.NewRequest("GET", "/", nil),
+				Resp: responseWriter,
+			},
+		}
+		store := mockstore.NewSQLStoreMock()
+
+		store.ExpectedPluginSetting = &models.PluginSetting{
+			SecureJsonData: map[string][]byte{},
+		}
```
Comment (marefr) [inline pkg/api/pluginproxy/pluginproxy_test.go:277]:
```suggestion
```

File: pkg/api/pluginproxy/pluginproxy_test.go [hunk 2]
```diff
@@ -245,6 +246,46 @@ func TestPluginProxy(t *testing.T) {
...
+		proxy := NewApiPluginProxy(ctx, "", route, "", &setting.Cfg{}, store, secretsService)
```
Comment (marefr) [inline pkg/api/pluginproxy/pluginproxy_test.go:278]:
```suggestion
		proxy := NewApiPluginProxy(ctx, "", route, "", &setting.Cfg{}, secretsService)
```

**Review by marefr — COMMENTED** [review]
Think these suggestions should resolve pluginproxy issues

**Comment by marefr** [comment]
Something with service accounts failing now that I don't know of. Is there something not backported to 8.4.x yet, or have something been merged to main targeting 8.5.x?

**Comment by Jguer** [comment]
Hey, seems to be failing on the changes of https://github.com/grafana/grafana/pull/44871 , how can I help?

**Comment by Jguer** [comment]
> Hey, seems to be failing on the changes of #44871 , how can I help?

File did not exist in 8.4, in back porting the patch, the full file as it existed in main was ported

**Comment by vtorosyan** [comment]
Thanks @Jguer and @marefr for jumping in.

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**